sdk/lib/drivers/sound/mmebuddy/wave/streaming.c

```diff
@@ -206,6 +212,16 @@ CompleteIO(
             /* Partially completed */
             HdrExtension->BytesCompleted += dwNumberOfBytesTransferred;
             SND_TRACE(L"%d/%d bytes of wavehdr completed\n", HdrExtension->BytesCompleted, WaveHdr->dwBufferLength);
+
+            /* Do we loop a header? */
+            if (HdrExtension->BytesCommitted == WaveHdr->dwBufferLength &&
+                !SoundOverlapped->PerformCompletion)
+            {
+                /* Reset amount of bytes and decrement loop count, to play next iteration */
+                HdrExtension->BytesCommitted = 0;
+                --SoundDeviceInstance->LoopsRemaining;
```

Comment:
> `0xFFFFFFFF` (`INFINITE` constant)

In this "edge" case, shouldn't the decrement be skipped?

Reply:
If `0xFFFFFFFF` can't be decremented, then yes, it should.

Reply:
Committed.